crd-docs/cr/kiali.io_v1alpha1_kiali.yaml

```diff
@@ -97,6 +97,8 @@ spec:
                 values:
                 - S2
             topologyKey: topology.kubernetes.io/zone
+    # default: cluster_wide_access is undefined
+    cluster_wide_access: false
```

Comment:
OK, so I am clear - cluster-wide-access will ALWAYS default to true if it is not specified (regardless of whether the user specifies AN or not).

I will document this here to register my uneasiness around that because now the behavior is different than before, and it is security related. Setting AN to non-** in previous versions ensured the user did not give kiali wide-open access. Now, by default, kiali is goign to be given wide open access regardless of AN.

If we are going to do this, we need to LOUDLY announce this change in the release notes.

Reply:
I'm not trying to force anything but what is the alternative?  If you default CWA to false, and also default AN to **, then that is an invalid config OOB.  At least that is what we said in the doc.  Otherwise we'd be granting namespace permission to all namespaces, right?  And I don't think we want that.

I do agree it needs to be documented clearly and we need to point out that you MUST set AN to your desired namespaces to allow for namespace-level permissions to those namespaces.

It's not too late to change our design, so let's hash it out.

Reply:
In the description of "cluster_wide_access" (CWA) that you looked at below, I had this:

```
When not specified, this value will be `false` if `deployment.accessible_namespaces`
does not have the special value of `'**'` otherwise this will be `true`.
```

So it isn't a fixed "CWA will always default to false" or "CWA will always default to true". It depends on the value of AN.
* If AN == ** then CWA will be true by default
* If AN != ** then CWA will be false by default

This is backward compatible -- that is how things worked in the past.

How I understand what you are asking for, you want cluster-wide-access to ALWAYS be true regardless of the value of AN. In that case, the second condition I show above will now have different behavior from before. It will now be:

* If AN != ** then CWA will still be true by default

So if AN != **, before users were used to getting a Kiali SA that was restricted to the namespaces in the list. Now it won't be - the Kiali SA will have cluster wide access.

The problem with the way I have this PR and how I just described it is: its complicated for users to grok (as is clear here because YOU didn't seem to understand how works). This is getting complicated with people having to understand how AN and CWA work in conjunction with each other.  And it doesn't even get into the whole excludes/includes/excludeSelector/includeSelector that we have now, too.

So, I'm not married to this. I am pointing out if we change this, we need to loudly mention this in the release notes and explain that the user must set "cluster-wide-access" to false in order to get what they had before.

Reply:
The issue is just for the folks updating, right?  Can we require CWA to be set, such that the operator updates the CR if it is unset, and CWA is set based on how AN is set? If AN has an explicit list, we set it to to false, otherwise true?  We are trying to untie CWA from AN, but maybe in an update scenario, where CWA is not set, we make a 1-time setting of the value?  That would maintain the existing behavior for current installs.  I dunno, it's late, we can talk about it tomorrow...

Reply:
> Can we require CWA to be set, such that the operator updates the CR if it is unset, and CWA is set based on how AN is set?

That's essentially how defaults work in the operator. The operator doesn't "modify" the CR except for the status fields, but it does set the ConfigMap to the appropriate default settings if the user doesn't specify something. The CWA default is set in the ConfigMap based on how AN is set. That just is how all the defaults are managed in the operator (if something isn't set, a default value is explicitly set in the ConfigMap). Remember, the server doesn't look at the CR at all - it only looks at the ConfigMap - the operator explicitly sets all values in the ConfigMap.

Reply:
OK, so after this and some off-PR discussion, I think we're now on the same page.   This file is just an example, so the `false` here is an example setting.  The important thing is in the comment line: ` # default: cluster_wide_access is undefined`, that by default CWA is unset, and we'll describe that behavior elsewhere.

Reply:
yes, exactly. Just an example and some times we document a non-default value (like here, but we do this elsewhere in this file, too), but when we do we always have a comment to tell the reader what the actual default is.

The point to this example file is it shows at a glance to the reader what the default values are for every setting. The settings in here (when not commented otherwise) are the actual default values.

We mention this at the top of the kiali.io page here: https://kiali.io/docs/configuration/kialis.kiali.io/#example-cr where we say,

```
(all values shown here are the defaults unless otherwise noted)
```

![image](https://user-images.githubusercontent.com/2029470/233448461-4710e284-408e-443f-b209-434faf39d9ca.png)

Reply:
Does the server config value need to be a bool pointer then where `not set == nil` or does it not matter at that point? I'm assuming the helm charts will have the same behavior as the operator?

Reply:
> Does the server config value need to be a bool pointer then where not set == nil or does it not matter at that point?

It doesn't matter because by the time the server gets the ConfigMap, all of these values will be guaranteed to be set. These docs are ONLY for the Kiali CR and does not correspond to the internal ConfigMap settings (this is why we tell  users not to mess with or worry about the ConfigMap - that is an implementation details users should not care about).

> I'm assuming the helm charts will have the same behavior as the operator?

No, the server helm chart does not have the same behavior as the operator wrt the accessible_namespaces feature. And no change is being made to the server helm chart at this point. The helm chart doesn't support creating Roles when `AN != **` and hasn't since the beginning. This is why we say this in the server helm chart values.yaml: https://github.com/kiali/helm-charts/blob/master/kiali-server/values.yaml#L29-L30
```
  # This only limits what Kiali will attempt to see, but Kiali Service Account has permissions to see everything.
  # For more control over what the Kial Service Account can see, use the Kiali Operator
  accessible_namespaces:
```

Reply:
I do want to mention that the server helm chart does NOT set cluster_wide_access (see https://github.com/kiali/helm-charts/blob/master/kiali-server/values.yaml), thus the server config value is not being set in the ConfigMap and so the bool value is falling back to the Go-default of `false`. So, I'm thinking the server PR probably needs to explicitly set cluster_wide_access to true in its defaults in config.go.

Either that or we create a helm chart PR to explicitly set cluster_wide_access to true in the server helm chart.